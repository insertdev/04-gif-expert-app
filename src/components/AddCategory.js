import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);

        // console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        // Evitar postback del form
        e.preventDefault();
        console.log('handleSubmit', inputValue);

        if (inputValue.trim().length > 2) {
            setCategories( cats => [inputValue,  ...cats]);

            // Resetea texto input
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
