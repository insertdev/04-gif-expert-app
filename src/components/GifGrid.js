import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);
    const {loading} = useFetchGifs();
    

    

    // // para que solo se ejecute una vez
    // // el array es necesario
    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages);
    // }, [category])

    return (
        <>
            <h3>{category}</h3>

            { loading ? 'Cargando...' : 'Data cargada' }
            {/* <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>  */}
        </>
    )
}
