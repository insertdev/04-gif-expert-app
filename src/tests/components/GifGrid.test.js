import React from 'react'
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('pruebas en <GifGrid />', () => {
    const category = 'Dragon Ball';
     
    test('debe de mostrarse correctamente', () => {
        // fingir resultado (mock)
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();

    })
    
    
})
