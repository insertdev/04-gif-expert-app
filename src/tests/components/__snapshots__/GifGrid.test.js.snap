// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
